soapui/src/main/java/com/eviware/soapui/impl/actions/RestServiceBuilder.java

```diff
@@ -64,27 +67,79 @@ public String getUri() {
         }
     }
 
-    public void createRestService(WsdlProject project, String URI) throws MalformedURLException {
+    public RestRequest createRestService(WsdlProject project, String URI) throws MalformedURLException {
+        return createRestService(project, URI, true);
+    }
+
+    public RestRequest createRestService(WsdlProject project, String URI, boolean showRequest) throws MalformedURLException {
         if (StringUtils.isNullOrEmpty(URI)) {
-            return;
+            throw  new MalformedURLException("The URL is null or empty");
         }
 
         RestResource restResource = createResource(ModelCreationStrategy.CREATE_NEW_MODEL, project, URI);
-        RestRequest restRequest = addNewRequest(addNewMethod(ModelCreationStrategy.CREATE_NEW_MODEL, restResource, RestRequestInterface.HttpMethod.GET));
+        RestRequest restRequest = addNewRequest(addNewMethod(ModelCreationStrategy.CREATE_NEW_MODEL,
+                restResource, RestRequestInterface.HttpMethod.GET));
         copyParameters(extractParams(URI), restResource.getParams());
-        UISupport.select(restRequest);
-        UISupport.showDesktopPanel(restRequest);
-
+        if (showRequest) {
+            UISupport.select(restRequest);
+            UISupport.showDesktopPanel(restRequest);
+        }
+        return restRequest;
     }
 
-    public RestRequest createRestServiceHeadlessFromUri(WsdlProject project, RequestInfo requestInfo, ModelCreationStrategy methodReuseStrategy) throws MalformedURLException {
+    public RestRequest createRestServiceHeadlessFromUri(WsdlProject project, RequestInfo requestInfo,
+                                                        ModelCreationStrategy methodReuseStrategy) throws MalformedURLException {
         RestResource restResource = createResource(REUSE_MODEL, project, requestInfo.getUri());
         RestMethod restMethod = addNewMethod(methodReuseStrategy, restResource, requestInfo.getRequestMethod());
         RestRequest restRequest = addNewRequest(restMethod);
         copyParametersWithDefaultsOnResource(extractParams(requestInfo.getUri()), restMethod.getParams(), restRequest.getParams());
         return restRequest;
     }
 
+    public RestRequest createRestServiceWithMethod(WsdlProject project, String URI,
```

Comment:
should this method be public, according to the signature it looks like private helper method.

Reply:
I use it in SaveRequestAction class